lib/galaxy/webapps/galaxy/api/library_datasets.py

```diff
@@ -647,6 +650,8 @@ def traverse(folder):
                         efp, fname = os.path.split(fpath)
                         if fname > '':
                             fname = fname.translate(trantab)
+                        if zpathext == '':
```

Comment:
I'd do this in all cases, otherwise any library item with a `.` in the name will be without the proper extension. I'd also simplify this to
`fname = "{fname}.{extension}".format(fname=fname, extension=ldda.extension)`
My inclination would be to also change that on line 632. This had been present since the initial commit, but I don't really see the reason for that. Do you have any objections to that @martenson ?